blog/_posts/2020-05-26-create-your-own-theme-detailed-walkthrough.md

```diff
@@ -0,0 +1,372 @@
+
+I use the *"eye dropper"* from [gimp](https://www.gimp.org/downloads/) :
+
+![](/public/screenshots/2020-05-26-gimp.png)
+
+I used it to click magically on the color that I needed to know the value...
+
+...And I get the html notation `#181818` that you will see later in the CSS : 
+
+![](/public/screenshots/2020-05-26-htmlnotation.png)
+
+
+## Panel backgrounds 
+
+Ok let's start changing the colors. 
+
+It is a **hacker** theme, it has to be daaark ! 
+
+Therefore I wanted to put black on left/mid panel and grey on right panel... I was actually not sure about how to
+arrange this, I made some tests and I finally used black everywhere !
+
+Then look at our first modification.
+
+``` /* Backgrounds */ --sidebar-left-bg: black; --body-bg: black; --sidebar-right-bg: black;
+
+```
+
+I can use color shortcuts names (`black`) or hex values (`#RRGGBB` from `#000000` to `#ffffff`)
+
+You can use for instance an [online color picker](https://www.w3schools.com/colors/colors_picker.asp) to find colors
+values or shades. 
+
+Let's imagine "I want a green like this one but darker" you can type the current green value in the color picker and
+choose another level of shade.
+
+![](/public/screenshots/2020-05-26-shades.png)
+
+
+## Edit text colors 
+
+Having such dark barckground immediatly revealed text colors issues. 
+
+On the left I will use a **hacker's green**, on the right a grey and in the middle a light green with higher contrast.
+
+```
+  /* Text color */
+  --text-color: #77ff77; /* The body text color (mid panel) */
+  --sidebar-right-text: #666666; /* Text color on right panel */
+  --sidebar-left-text: green; /* Color of rooms when connected (selected or not) */
+```
+
+![](/public/screenshots/2020-05-26-textcolor.png)
+
+
+## Links
+
+Next we can change the links color (send button, upload file, settings) to green (it is the hacker color that we will
+use with dark shades) : 
+
+`--link-color: green;`
+
+will produce :
+
+![](/public/screenshots/2020-05-26-l1.png)
+
+![](/public/screenshots/2020-05-26-l2.png)
+
+![](/public/screenshots/2020-05-26-l3.png)
+
+![](/public/screenshots/2020-05-26-l4.png)
+
+Next, the highlight when you are mentionned. I will choose a red to hightlight when someone talk to me : 
+
+`--highlight-color-bg: #990000;`
+
+I do not use `red` color name because it is too flashy.
+
+![](/public/screenshots/2020-05-26-hl.png)
+
+
+## Left side panel
+
+We will try to put a lot of green here.
+
+The more important here will be the shadow around the room when selected.
+
+It is achieved with ` --focus-outline: 0 0 3px 3px rgba(0, 255, 0, 0.5);` where the *rgba* is clearly "give me a shadow
+of green".
+
+![](/public/screenshots/2020-05-26-shadow.png)
+
+Then I set the `--sidebar-left-border: 1px solid black;` (which is an invisible border between left panel and mid panel.  
+
+Then I set the `--sidebar-left-search-focus-bg: black;` that will let an impression of transparency (simply because we
```

Comment:
You can also use [transparent](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#transparent_keyword).

Reply:
Changed to transparent